### PR TITLE
#139 Convert demos to stories

### PR DESCRIPTION
**Pull request by rpeterman-gp**
All demos have been converted to stories, with a few caveats.

- Global CSS stories are basically straight ports of demo version as these will need to be refactored when moving global styles into applicable components. As for previewing changes to these styles, the stories will facilitate this need.
- Auth, Guard, and Hal demos are awkward to represent as stories since their functionality is dependent on running in an environment that supports proper authentication (SSL, proxied domain, etc.). Also, the current state of the docs for these services are sparse, hard to follow without context, and their functional demo code doesn't reflect what is instructed or is missing functional example for some instructions. Help from someone with more expertise in usage of these service would be appreciated.
- BaseModel story is pretty much just documentation.
- Navigation related stories whose components rely on router directives (Header stories, Tabs) needed to be setup with routing that can interfere with storybooks hot-reload of content iframe (Storybook's iframe app doesn't usually have routing setup.)

### CODE REVIEW

File: projects/ngx-prx-styleguide/src/lib/spinner/spinner.component.stories.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    {
+      notes: {
+        markdown:`
+# Spinner
+
+The Spinner Component shows a loading indicator. Typically the spinner is
+displayed with \`*ngIf\` until a resource has finished loading.
+
+The animation uses \`position: absolute\`, so it should be inside a
+\`position: relative\` container.
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/src/lib/spinner/spinner.component.stories.ts]: This `position: relative` needs to be in freakin bold or better yet delegated to the spinner component. I forget about it more often than I remember even though it's just basic positioning.

File: projects/ngx-prx-styleguide/src/assets/styles/stories/button.stories.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+  .addDecorator(centered)
+  .addDecorator(storiesModuleMetaData)
+  .add(
+    'Button',
+    () => ({
+      styleUrls: [
+        '../../../../../../src/styles.css'
+      ],
+      template: `
+        <section class="main demo">
```
[the remark this answers is not in the document]
Comment (rpeterman-gp) [reply projects/ngx-prx-styleguide/src/assets/styles/stories/button.stories.ts]: Yes, that is the plan. I only continued using the demo css and markup here since these stories are acting as a test bed for tweaks to legacy css. In the future I want to flush out the stories for the global CSS to be a little more targeted as things are refactored.

File: projects/ngx-prx-styleguide/src/assets/styles/stories/forms.stories.ts [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+      ],
+      template: `
+        <section class="main demo">
+          <h1>Forms</h1>
+          <p>A style sheet for form element look and feel</p>
+
+          <section>
+            <h2>Form</h2>
+            <ul>
+              <li><code>display: flex</code> with the following flex properties</li>
```
[the remark this answers is not in the document]
Comment (rpeterman-gp) [reply projects/ngx-prx-styleguide/src/assets/styles/stories/forms.stories.ts]: Yeah. I thought this was weird. This like a layout decision that should be left to the applications.
